examples/server/server.cpp

```diff
@@ -1574,6 +1599,13 @@ struct server_response {
 
         std::unique_lock<std::mutex> lock(mutex_results);
         waiting_task_ids.erase(id_task);
+        // make sure to clean up all pending results
+        for (size_t i = 0; i < queue_results.size(); i++) {
+            if (queue_results[i]->id == id_task) {
+                queue_results.erase(queue_results.begin() + i);
+                i--;
+            }
+        }
```

Comment:
Either use `std::remove_if` or avoid the decrement of unsigned integer like this:

```suggestion
        for (size_t i = 0; i < queue_results.size(); ) {
		    if (queue_results[i]->id == id_task) {
		        queue_results.erase(queue_results.begin() + i);
		    } else {
		        i++;
		    }
		}
```